spec/tracking_spec.js

```diff
@@ -0,0 +1,92 @@
+describe('App Tracking', function() {
+  var tracking = require('tracking'),
+      sandbox  = sinon.sandbox.create(),
+      client = { invoke: function() {} },
+      eventListener;
+
+  beforeEach(function() {
+    sandbox.stub(client, 'invoke');
+    eventListener = sinon.spy(window, 'addEventListener');
+    tracking.setup(client);
+  });
+
+  afterEach(function() {
+    window.addEventListener.restore();
+    sandbox.restore();
+  });
+
+  function findEvent(event) {
+    for(var i = 0; i < eventListener.args.length; i++) {
+      if(eventListener.args[i][0] === event) {
+        return eventListener.args[i];
+      }
+    }
+  }
```

Comment:
This is fragile. It's possible for multiple listeners of a single type to be defined. This method might grab the wrong one and cause tests to fail despite the code working fine. I might experiment with looking for events based on function name (you'd need to give them function names with `function hover(a,b,c) {...}`). Or not test the event registration at all. Instead pulling the callbacks out into their own named functions and testing their logic as unit tests with none of the need for the `beforeEach` and `afterEach` or `findEvent` stuff that you have here.